### ISSUE TITLE
Human friendly format

### ISSUE THREAD

**Issue by nick11337** [smaxtec/adis#5]
Hey,

what about an implementation of a "string_value". For example:

`
"880028": {
        "definitions": [
            {
                "item_number": "00800004",
                "field_size": 15,
                "decimal_digits": 0,
                "string_value": "Betriebsnummer"
            },
            {
                "item_number": "00800043",
                "field_size": 2,
                "decimal_digits": 0,
                "string_value": "Betriebsstätte (AE)"
            }
`

I already wrote a script that scrapes the data from http://ian.lkv-nrw.de (just german) and inserts the field. I am not that into ADIS files, but is that a good idea?

**Comment by Philipp1221** [comment]
Hey nick11337, thanks for your issue.
The thing is that adis just specifies the Interchange Syntax, the data mapping format (Data Element Dictionary) is specified in the aded part and as far as I know, there is a global standard for this but no one uses this standard and that is why we just implemented the adis part.

**Comment by nick11337** [comment]
Ah okay, thanks for your answer. Also great work, exactly yesterday we needed some kind of libary like this. I wil modify it. If you want we can stay in exchange.
Otherwise i have a mini mini PR, i will send it to you :)

**Comment by Philipp1221** [comment]
If there is anything to improve or fix please don't hesitate and just open an issue or write us we are happy for every contribution :)

**Comment by smaXtecStefan** [comment]
I had the same thing in my mind. It is tedious to always look up what the numbers mean, when you are filling the json or trying to read the adis file.

My suggestion would be that the string is then in the file as comment.

Input (`entity_name` and `item_name` added):

```json
[
    {
        "990001": {
            "entity_name": "natural_constants",
            "definitions": [
                {
                    "item_number": "00900001",
                    "item_name": "constant",
                    "field_size": 20,
                    "decimal_digits": 0
                },
                {
                    "item_number": "00900002",
                    "item_name": "value",
                    "field_size": 9,
                    "decimal_digits": 6
                },
                {
                    "item_number": "00900003",
                    "item_name": "unit",
                    "field_size": 10,
                    "decimal_digits": 0
                }
            ],
            "data": [
                {
                    "00000000": "Gravity on Earth    ",
                    "00000001": 9.81,
                    "00000002": "ms^(-2)   "
                }
            ],
            "status": "N"
        }
    }
]
```

created adis file (`entity_name` and `item_name` are in the comments):

```adis
CN natural_constants
CN      |constant  |value     |unit
DN990001009000012000090000209600900003100
CN      |constant             |value |unit
VN990001Gravity on Earth      9810000ms^(-2)   
ZN
```

**Comment by nick11337** [comment]
So you assume that the adis file has the comments already inside it?
I think this could be quite difficult. @Philipp1221 explained to me that the data dictionary may be different for every region/state, whatever. So either we have the `item_name` in the ADIS-File or we need to scrape them based on some fields. 

For example:
In my ADIS File there is a number translated to "Datenherkunft", in german sth like region. In my case it is NRW, so we need to get the mapping list for NRW. For my Use Case I extract the dictionary from their webpage and translate them to get a clean and nice csv.

![image](https://user-images.githubusercontent.com/64522410/133785897-b44fe8fd-edb9-46c5-bf23-b7085b46d549.png)

Feel free to comment on my use-case.

Unfortunately i am on holidays next week, so i won't be able to communicate with you until the week after.

**Comment by smaXtecStefan** [comment]
I think I get now what you mean.

You have an adis file and want to convert it to json. (This is what the library already does.)
But in addition you would like to have a string which describes the item_number in the definitions part. Am I right?

Input:

```adis
DN8800280080000415000800043020
ZN
```

Output:

```json
[
    {
        "880028": {
            "definitions": [
                {
                    "item_number": "00800004",
                    "field_size": 15,
                    "decimal_digits": 0,
                    "string_value": "Betriebsnummer"
                },
                {
                    "item_number": "00800043",
                    "field_size": 2,
                    "decimal_digits": 0,
                    "string_value": "Betriebsstätte (AE)"
                }
            ],
            "data": [],
            "status": "N"
        }
    }
]
```

I could imaging to provide a mapping, which adds the description of the item_number:

``` python
mapping = {
    "00800004": "Betriebsnummer",
    "00800043": "Betriebsstätte (AE)"
}
Adis.parse_from_file("sample.ads", mapping)
```

Would this be helpful?

**Comment by nick11337** [comment]
Yeah exactly. I already did this in my code with extracting the data from the web. Should i do a small PR and u take a look @ the code? It was just a fast implementation, but maybe this helps :) I saw you just added a new issue for that, should we continue there?

**Comment by smaXtecStefan** [comment]
Sure, you can create a PR :) 
I would see those two issues as two different:

This issue here as adding a mapping which translates the item_number to item_name.
The other ([Add names to json and comment lines #7 ](https://github.com/smaxtec/adis/issues/7)) as adding comments to the adis file from the item_name.

**Comment by nick11337** [comment]
So i created a PR, but i don't know how to link it with this issue.

https://github.com/smaxtec/adis/pull/11 This is the PR :)
